### ISSUE TITLE
IPInfo account is being abused

### ISSUE THREAD

**Issue by cjyabraham** [kubernetes/website#40928]
The search on this site [uses ipinfo.io to determine whether a user is in China or not](https://github.com/kubernetes/website/blob/main/static/js/search.js#L72). The usage of this account went through a spike last month which cost CNCF hundreds of dollars. Most likely this isn't because of usage from this site but because someone has stolen our token and is using it for other purposes. Note that the token is just [exposed in the code](https://github.com/kubernetes/website/blob/main/static/js/search.js#L74) and downloaded by all site visitors. I suggest we find another way to perform this determination so that we can remove this vulnerability and stop wasting money. Recent usage of ipinfo.io is below:

![chart](https://user-images.githubusercontent.com/463160/235644372-93f560b9-48d4-4b3c-9263-19def0eb92ad.png)

**Comment by tamilselvan1102** [comment]
Kubernetes website pages are building as static pages so all the API calls happening through JavaScript and hence there is no other way to hide the token.

**Comment by sftim** [comment]
We might be able to use a Netlify Function instead.

**Comment by cjyabraham** [comment]
BTW, I'm curious, do you have a sense of what proportion of visitors inside of China require this functionality and what proportion have a VPN so can access Google search results? I'm asking to determine to what extent we need to provide Chinese-based alternatives for other CNCF resources.

**Comment by cjyabraham** [comment]
It turns out there is a way to whitelist referring domains on ipinfo. I have added kubernetes.io to the list and I assume this will take care of any abuse issues, so think we can close this issue if people are happy with this solution.

I would still, however, be interested in a reply to my question in the previous comment.

**Comment by sftim** [comment]
Does that allow list also cover v1-26.docs.kubernetes.io etc @cjyabraham ?

**Comment by sftim** [comment]
I don't think we have an insight into what proportion of people inside China use a VPN to access our website. AIUI people not using a VPN, and connecting from China, don't appear in our Google Analytics stats.

**Comment by cjyabraham** [comment]
I added `kubernetes.io` to the whitelist so all subdomains should be included. We should test this out to make sure they're all still working.

**Comment by sftim** [comment]
The subdomains use the main search page anyway.

Done.
/close
